src/commands/gate/evaluate.ts

```diff
@@ -169,20 +169,21 @@ export class GateEvaluateCommand extends Command {
    * - If the request is successful, the promise will be resolved with the response
    * - If the request is successful but the status is 'wait', the promise will be rejected after the received wait time (wait_time_ms)
    * - If the request is not successful, the promise will be rejected after `initialRetryMs`, with an exponential factor that depends on the attempt (exponential backoff).
-   * - If the command execution time is greater than the command timeout, the promise will be rejected immediately
+   * - If the command execution time is greater than the command timeout, we bail to avoid retrying.
    * If the promise is rejected, `retryRequest` will handle the retry immediately.
    */
   private async evaluateRulesWithWait(
     api: APIHelper,
     evaluateRequest: Payload,
-    attempt?: number
+    attempt?: number,
+    bail?: (e: Error) => void
   ): Promise<AxiosResponse<EvaluationResponsePayload>> {
     const timePassed = new Date().getTime() - evaluateRequest.startTimeMs
     const remainingWait = this.timeoutInSeconds * 1000 - timePassed
 
     return new Promise((resolve, reject) => {
       if (remainingWait <= 0) {
-        reject(new Error('wait'))
+        bail?.(new Error('wait'))
```

Comment:
do we want to bail out with `'wait'` here? There's nothing to wait for, right?

Reply:
It was more convenient, because we already process `wait` errors. I can change the bail to throw `new Error('timeout')`, and process `timeout` errors as well, wdyt?

Reply:
but we don't want to process wait errors the same way, right?

Reply:
In`retryRequest` catch it's the same I think, It can happen if we exhaust retries and the response from the service is  still `wait`.